src/api.js

```diff
@@ -23,14 +23,16 @@ const API = {
       'onRequestCompleted',
       'requestData',
       'requestHeaders',
-      'urlRoot'
+      'urlRoot',
+      'superagent'
     ]))
   },
-	
+
 	request(options = {}) {
 
 		let { method, data, endpoint, onSuccess, onError, fileData } = options;
     let requestData, requestHeaders, doRequest;
+    const request = this.superagent || superagent;
```

Comment:
you have to pass superagent instance in options first and include it in destructing assignment, then assign `this.superagent` variable